# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/TabChild.h
# Commit: 1ba9a2cebf12
# Full Hash: 1ba9a2cebf1252ffc1af89e3e978cf4b47fcb46e
# Author: Barret Rennie <barret@brennie.ca>
# Date: 2019-04-04 04:33:36
# Regressor Bug: 1510569
# File Overlap Count: 1
# Description:
#   Bug 1510569 - Filter TabChild nsIWebProgress events through a nsBrowserFilter r=kmag
#   
#   The implementation of WebProgressChild.jsm filters all nsIWebProgress events
#   through an nsBrowserFilter. We need to do this in the TabChild as well, but
#   TabChild currently only handles onContentBlockingEvent, for which
# ==============================================================================

diff -r a928d7bd9427 -r 1ba9a2cebf12 dom/ipc/TabChild.h
--- a/dom/ipc/TabChild.h	Wed Apr 03 17:28:56 2019 +0000
+++ b/dom/ipc/TabChild.h	Wed Apr 03 17:29:15 2019 +0000
@@ -45,6 +45,7 @@
 #include "nsDeque.h"
 #include "nsISHistoryListener.h"
 
+class nsBrowserStatusFilter;
 class nsIDOMWindowUtils;
 class nsIHttpChannel;
 class nsIRequest;
@@ -804,6 +805,7 @@
   nsCOMPtr<nsIURI> mLastURI;
   RefPtr<ContentChild> mManager;
   RefPtr<BrowsingContext> mBrowsingContext;
+  RefPtr<nsBrowserStatusFilter> mStatusFilter;
   uint32_t mChromeFlags;
   uint32_t mMaxTouchPoints;
   layers::LayersId mLayersId;
@@ -820,7 +822,7 @@
   SetAllowedTouchBehaviorCallback mSetAllowedTouchBehaviorCallback;
   bool mHasValidInnerSize;
   bool mDestroyed;
-  bool mProgressListenerRegistered;
+
   // Position of client area relative to the outer window
   LayoutDeviceIntPoint mClientOffset;
   // Position of tab, relative to parent widget (typically the window)
